File: app/src/main/java/org/mesonet/app/userdata/Preferences.java
package org.mesonet.app.userdata;

import java.util.Observable;



public interface Preferences
{
    enum UnitPreference{ kMetric, kImperial }



    UnitPreference GetUnitPreference();
    void SetUnitPreference(UnitPreference inPreference);
    String GetSelectedStid();
    void SetSelectedStid(String inStid);
    String GetSelectedRadar();
    void SetSelectedRadar(String inRadarName);
    PreferencesObservable GetPreferencesObservable();
}

File: app/src/main/java/org/mesonet/app/userdata/Preferences.kt
package org.mesonet.app.userdata

import java.util.Observable


interface Preferences {
    enum class UnitPreference {
        kMetric, kImperial
    }


    fun GetUnitPreference(): UnitPreference
    fun SetUnitPreference(inPreference: UnitPreference)
    fun GetSelectedStid(): String
    fun SetSelectedStid(inStid: String)
    fun GetSelectedRadar(): String
    fun SetSelectedRadar(inRadarName: String)
    fun GetPreferencesObservable(): PreferencesObservable
}